src/main/scala/cromwell/engine/db/slick/FailureEventComponent.scala

```diff
@@ -0,0 +1,49 @@
+package cromwell.engine.db.slick
+
+import java.sql.{Timestamp, Clob}
+
+import cromwell.engine.db.ExecutionDatabaseKey
+import cromwell.engine.{WorkflowId, ExecutionIndex}
+import cromwell.engine.ExecutionIndex._
+
+case class FailureEvent (
+  workflowExecutionId: Int,
+  executionId: Option[Int],
+  failure: String,
+  timestamp: Timestamp,
+  failureId: Option[Int] = None
+)
+
+trait FailureEventComponent {
+  this: DriverComponent with ExecutionComponent with WorkflowExecutionComponent =>
+
+  import driver.api._
+
+  class FailureEvents(tag: Tag) extends Table[FailureEvent](tag, "FAILURE") {
+    def failureId = column[Int]("FAILURE_ID", O.PrimaryKey, O.AutoInc)
+    def workflowExecutionId = column[Int]("WORKFLOW_EXECUTION_ID")
+    def executionId = column[Option[Int]]("EXECUTION_ID")
+    def failure = column[String]("FAILURE")
+    def timestamp = column[Timestamp]("TIMESTAMP")
+
+    override def * = (workflowExecutionId, executionId, failure, timestamp, failureId.?) <>
+      (FailureEvent.tupled, FailureEvent.unapply)
+
+    def workflowExecution = foreignKey("FK_FAILURE_WORKFLOW_EXECUTION_ID", workflowExecutionId, workflowExecutions)(_.workflowExecutionId)
+    def execution = foreignKey("FK_FAILURE_EXECUTION_ID", executionId, executions)(_.executionId)
+  }
+
+  protected val failureEvents = TableQuery[FailureEvents]
+
+  val failureEventsAutoInc = failureEvents returning failureEvents.
+    map(_.failureId) into ((a, id) => a.copy(failureId = Some(id)))
```

Comment:
someone might ask you to use `Option(id)` though the id *really* better not be null